fdbsyncd/fdbsync.cpp

```diff
@@ -0,0 +1,603 @@
+		}
+		else
+		{
+			type = "static";
+		}
+
+		const std::string cmds = std::string("")
+			+ " bridge fdb " + "add" + " " + kmac + " dev "
+			+ port_name + " master " + type  + " vlan " + to_string(vlan);
+
+		std::string res;
+		int ret = swss::exec(cmds, res);
+		if (ret != 0)
+		{
+			SWSS_LOG_INFO("Failed cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+		}
+
+		SWSS_LOG_NOTICE("Refreshing cmd:%s, res=%s, ret=%d", cmds.c_str(), res.c_str(), ret);
+    }
+    return;
+}
+
+bool FdbSync::checkImetExit(string key, uint32_t vni)
+{
+    if (imet_route[key].redist_add_status == true)
+    {
+        SWSS_LOG_DEBUG("IMET exist key:%s Vni:%d", key.c_str(), vni);
+        return false;
+    }
+    imet_route[key].vni =  vni;
+    return true;
+}
+
+bool FdbSync::checkDelImet(string key, uint32_t vni)
+{
+    int ret = true;
+
+    SWSS_LOG_DEBUG("Del IMET key:%s Vni:%d", key.c_str(), vni);
+    if (imet_route.find(key) != imet_route.end())
+    {
+        SWSS_LOG_DEBUG("IMET found key:%s Vni:%d", key.c_str(), vni);
+        if (imet_route[key].redist_add_status == false)
+        {
+            SWSS_LOG_DEBUG("IMET found key:%s Vni:%d, it should have not happened", key.c_str(), vni);
+            ret = false;
+        }
+        imet_route.erase(key);
+    }
+    return ret;
+}
+
+void FdbSync::imetAddRoute(struct in_addr vtep, string vlan_str, uint32_t vni)
+{
+    string vlan_id = "Vlan" + vlan_str;
+    string key = vlan_id + ":" + inet_ntoa(vtep);
+
+    if (!checkImetExit(key, vni))
+        return;
+
+	SWSS_LOG_NOTICE("%sIMET Add route key:%s vtep:%s %s", 
+			m_AppRestartAssist->isWarmStartInProgress() ? "WARM-RESTART:" : "",
+			key.c_str(), inet_ntoa(vtep), vlan_id.c_str());
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple f("vni", to_string(vni));
+    fvVector.push_back(f);
+
+    imet_route[key].redist_add_status = true;
+    
+	// If warmstart is in progress, we take all netlink changes into the cache map
+    if (m_AppRestartAssist->isWarmStartInProgress())
+    {
+        m_AppRestartAssist->insertToMap(APP_EVPN_REMOTE_VNI_TABLE_NAME, key, fvVector, false);
+        return;
+    }
+    
+	m_imetTable.set(key, fvVector);
+    return;
+}
+
+void FdbSync::imetDelRoute(struct in_addr vtep, string vlan_str, uint32_t vni)
+{
+    string vlan_id = "Vlan" + vlan_str;
+    string key = vlan_id + ":" + inet_ntoa(vtep);
+
+    if (!checkDelImet(key, vni))
+        return;
+
+	SWSS_LOG_NOTICE("%sIMET Del route key:%s vtep:%s %s", 
+			m_AppRestartAssist->isWarmStartInProgress() ? "WARM-RESTART:" : "", 
+			key.c_str(), inet_ntoa(vtep), vlan_id.c_str());
+
+    std::vector<FieldValueTuple> fvVector;
+    FieldValueTuple f("vni", to_string(vni));
+    fvVector.push_back(f);
+
+    // If warmstart is in progress, we take all netlink changes into the cache map
+    if (m_AppRestartAssist->isWarmStartInProgress())
+    {
+        m_AppRestartAssist->insertToMap(APP_EVPN_REMOTE_VNI_TABLE_NAME, key, fvVector, true);

```

Comment:
According to sonic-swss-common [PR339](https://github.com/Azure/sonic-swss-common/pull/339), should this constant be renamed to `APP_VXLAN_REMOTE_VNI_TABLE_NAME`

Reply:
Thanks for review. Updated the same